components/icm42670/icm42670.c

```diff
@@ -414,3 +415,71 @@ esp_err_t icm42670_complimentory_filter(icm42670_handle_t sensor, const icm42670
 
     return ESP_OK;
 }
+
+esp_err_t icm42670_read_register(icm42670_handle_t sensor, uint8_t reg, uint8_t *val)
+{
+    return icm42670_read(sensor, reg, val, 1);
+}
+
+esp_err_t icm42670_write_register(icm42670_handle_t sensor, uint8_t reg, uint8_t val)
+{
+    return icm42670_write(sensor, reg, &val, 1);
+}
+
+esp_err_t icm42670_read_mreg_register(icm42670_handle_t sensor, uint8_t mreg, uint8_t reg,
+                                       uint8_t *val)
+{
+    uint8_t blk_sel_r = 0;
+    if (mreg == 1) {
+        blk_sel_r = 0;
+    } else if (mreg == 2) {
+        blk_sel_r = 0x28;
+    } else if (mreg == 3) {
+        blk_sel_r = 0x50;
+    } else {
+        ESP_LOGE(TAG, "Invalid MREG value %d", mreg);
+        return ESP_ERR_INVALID_ARG;
+    }
+    ESP_RETURN_ON_ERROR(icm42670_write(sensor, ICM42670_BLK_SEL_R, &blk_sel_r, 1), TAG,
+                        "Failed to set BLK_SEL_R");
+    ESP_RETURN_ON_ERROR(icm42670_write(sensor, ICM42670_MADDR_R, &reg, 1), TAG,
+                        "Failed to set MADDR_R");
+
+    int64_t start = esp_timer_get_time();
+    while (esp_timer_get_time() - start < 10000) {
```

Comment:
Could you use this `vTaskDelay(pdMS_TO_TICKS(10));` instead of the while loop?

Reply:
My mistake, this is supposed to be >10µs according to the datasheet. The spin loop is probably more correct for something that short right? 
<img width="682" alt="image" src="https://github.com/user-attachments/assets/474525b1-a24f-4b13-b7a2-3b61d64d13c5" />

Reply:
For 10 us, it is ok. Thank you for clarification.